src/libmongoc/src/mongoc/CMakeLists.txt

```diff
@@ -74,6 +74,7 @@ extra_dist_generated (
 )
 
 set (src_libmongoc_src_mongoc_DIST_noinst_hs
+   mcd-azure.h
```

Comment:
I would prefer the latter, if only because my tools better understand that the `.c` and `.h` files are related 😃 so I can automatically jump between them with a hotkey.

In fullness, I'd prefer private headers to be namespaced with a subdirectory, but that may require additional CMake caches.